mypy/plugins/dataclasses.py

```diff
@@ -413,6 +413,13 @@ def collect_attributes(self) -> Optional[List[DataclassAttribute]]:
                         if attr.name == name:
                             all_attrs.remove(attr)
                             super_attrs.append(attr)
+                            # When we have an override,
+                            # we need to check its compatibility.
+                            self._validate_field(
+                                attr,
+                                found_default=data.get("has_default", False),
+                                found_kw_sentinel=False,
```

Comment:
Are there any tests yet that check what happens if you have kw_only in a base class but not a child class, and variations around that?